### PR TITLE
json output parial implementation

### PR DESCRIPTION
**Pull request by davidlunarg**
Add json output, plus some miscellaneous fixes

### CODE REVIEW

File: framework/decode/vulkan_replay_resource_dump_json.h [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  // a separate method because VulkanReplayResourceDump is calling exit(0),
+  // which results in destructors not being called and the json file not getting
+  // closed out. VulkanReplayResourceDump explicity calls this method before
+  // calling exit(0).
+  void VulkanReplayResourceDumpJsonClose() {
+    if (json_writer_) {
+      json_writer_->EndStream();
+      gfxrecon::util::platform::FileClose(jsonFileHandle_);
+
+      //@@@ TODO: Not sure if this is the best way to release these. Could I
```
Comment (panos-lunarg) [inline framework/decode/vulkan_replay_resource_dump_json.h:97]: Do these variables need to be pointers? If so perhaps they can be smart pointers (unique_ptr) so they will be deleted by themselves once the destructor is called?

File: framework/decode/CMakeLists.txt [hunk 2]
```diff
@@ -254,13 +254,15 @@ endif()
 if (WIN32)
 target_include_directories(gfxrecon_decode
                            PUBLIC
+                               ${CMAKE_BINARY_DIR}
```
Comment (panos-lunarg) [inline framework/decode/CMakeLists.txt:257]: Is this necessary?

File: framework/decode/CMakeLists.txt [hunk 3]
```diff
@@ -254,13 +254,15 @@ endif()
...
                                ${CMAKE_SOURCE_DIR}/framework
                                ${CMAKE_SOURCE_DIR}/external
                                ${CMAKE_SOURCE_DIR}/external/precompiled/win64/include/
                                $<$<BOOL:${D3D12_SUPPORT}>:${CMAKE_SOURCE_DIR}/external/AgilitySDK/inc>)
 else()
 target_include_directories(gfxrecon_decode
                            PUBLIC
+                               ${CMAKE_BINARY_DIR}
```
Comment (panos-lunarg) [inline framework/decode/CMakeLists.txt:265]: Is this necessary?

**Comment by davidlunarg** [comment]
Abandoning. Replaced with pull request #16 from https://github.com/davidlunarg/gfxreconstruct/tree/davidp_dr_dev_json_0220)

**Pull request closed without merging**